src/Metadata/Api/HookMethods.php

```diff
@@ -53,55 +58,76 @@ public function hookMethods(string $className): array
 
             if ($method->isStatic()) {
                 if ($metadata->isBeforeClass()->isNotEmpty()) {
-                    array_unshift(
-                        self::$hookMethods[$className]['beforeClass'],
-                        $methodName,
+                    $beforeClass = $metadata->isBeforeClass()->asArray()[0];
+                    assert($beforeClass instanceof BeforeClass);
+
+                    self::$hookMethods[$className]['beforeClass']->prepend(
```

Comment:
because some of the hooks were added with `array_unshift`  and other ones were added at the end of the array, I introduced `prepend()` and `add()` methods, in order to keep the initial order when all hooks have default priority